raiden/transfer/architecture.py

```diff
@@ -138,3 +138,23 @@ def dispatch(self, state_change):
         assert all(isinstance(e, Event) for e in events)
 
         return events
+
+    def __eq__(self, other):
+        if isinstance(other, StateManager):
```

Comment:
these implementations are too complex / hard to understand. Let's use something more simpler like:
```
self.current_state == other.current_state
self.state_transition == other.state_transition
```